### PR TITLE
Fix: Make script birthdates dynamic

### PR DESCRIPTION
**Pull request by krisrjohnson21**
This PR updates our Youth Pass automation scripts to use dynamic birthdate ranges. These dynamic ranges replace hard-coded year values which would otherwise require manual updates in future years. I also noticed some odd spacing in the birthdate values that could potentially cause false failures, so I added simple regex to remove spacing.

Asana ticket: https://app.asana.com/0/1170867711449497/1201244803227001/f

### CODE REVIEW

File: cypress/automated-tests/youth-pass/address-section-required-fields.spec.js [hunk 1]
```diff
@@ -4,8 +4,14 @@ describe('Youth Pass Address Section Required Fields', () => {
     it('proceeds through an application', () => {
         const applicantZipCode = '02114';
         const youthPassUrl = Cypress.env('youth_pass_url');
-        const randomBirthdate = faker.date.between('2003-11-02', '2009-11-01');
-        const applicantBirthdate = `${randomBirthdate.getMonth() + 1}/${randomBirthdate.getDate()}/${randomBirthdate.getFullYear()}`;
+        const todaysDate = new Date();
+        const twelveYearsAgo = todaysDate.getFullYear() - 12; 
+        const eighteenYearsAgo = todaysDate.getFullYear() - 18; 
+        const randomBirthdate = faker.date.between(`${eighteenYearsAgo}-11-02`, `${twelveYearsAgo}-11-01`);
```
Comment (arkadyan) [inline cypress/automated-tests/youth-pass/address-section-required-fields.spec.js:10]: suggestion: A bunch of these constants and functions seem repeated across test files (and I could imagine them being useful for even more tests in the future. What would you think about consolidating them into a helpful file and tests could import whatever they needed?

File: cypress/automated-tests/youth-pass/address-section-required-fields.spec.js [hunk 2]
```diff
@@ -4,8 +4,14 @@ describe('Youth Pass Address Section Required Fields', () => {
...
+        const applicantBirthdate = 
+            `${randomBirthdate.getMonth() + 1}/
+            ${randomBirthdate.getDate()}/
+            ${randomBirthdate.getFullYear()}`.replace(/\s+/g, "");
```
Comment (arkadyan) [inline cypress/automated-tests/youth-pass/address-section-required-fields.spec.js:14]:
```suggestion
        const applicantBirthdate =  `${randomBirthdate.getMonth() + 1}/${randomBirthdate.getDate()}/${randomBirthdate.getFullYear()}`;
```

suggestion: I suspect the spacing you are seeing in the values is due to the line breaks here, but if you keep your string literal on a single line I think you might not need to go through the replacement step.

**Review by arkadyan — CHANGES_REQUESTED** [review]
I added a couple of suggestions that I think will make this all even simpler if you think they make sense. I'm excited to see this happening though because it means one less reason for a test to suddenly fail on us in a year.

File: cypress/automated-tests/youth-pass/address-section-required-fields.spec.js (hunk 1 above)
Comment (krisrjohnson21) [reply cypress/automated-tests/youth-pass/address-section-required-fields.spec.js:10]: Love this suggestion!

File: cypress/automated-tests/youth-pass/address-section-required-fields.spec.js (hunk 2 above)
Comment (krisrjohnson21) [reply cypress/automated-tests/youth-pass/address-section-required-fields.spec.js:14]: Got it. I was trying to respect line length here, but it's much simpler to keep these literals on a single line

**Review by arkadyan — APPROVED** [review]
👍

**Pull request merged**